### DISCUSSION TITLE
Quick command buffer?

### DISCUSSION THREAD

**Discussion by okuma10** [wezterm/wezterm#3863] [Q&A] [answered]
Wondering if one can create a "macro" or a hotkey to toggle directory changes?
Basically a way to explicitly select or toggle between commands in some buffer. 
Like if I'm working on a project X at directory `<starting work dir>/projects/<project X>`.  I have to type the typical `cd <starting work dir>/projects/<project X>` to get there. 
But let's say in the course of working on other stuff as well I add other commands, then for whatever reason I close wezterm. Next time I open it...if I don't want to write that again I have to keep pressing UP until I reach that command again. So I thought it would be nice if I could toggle between some kind of buffer that holds several quick commands like dir change. In a sense like the Harpoon addon for neovim, where you could toggle with TAB or some key combo to quickly open a buffer/or the file(if not in a buffer already).

**Comment by bew** [comment, marked as answer]
I'd suggest a solution using your shell', not the terminal.

Personally I use [zoxide](https://github.com/ajeetdsouza/zoxide), combined with fzf to fuzzy-jump anywhere I've been at least once by typing a few letters.
There are many other solutions, and making your own isn't too complex either.
There's also `cdr` on zsh [that you can enable](https://man.archlinux.org/man/zshcontrib.1#REMEMBERING_RECENT_DIRECTORIES), that remembers the last N directories (cross shell instances) and allow you to get back easily

**Comment by okuma10** [comment]
thanks I think that's what I was looking for
